kamu-core/tests/tests/test_reset_service_impl.rs

```diff
@@ -0,0 +1,202 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use kamu::domain::*;
+use kamu::infra::*;
+use kamu::testing::*;
+use opendatafabric::*;
+use tempfile::TempDir;
+
+
+#[test_log::test(tokio::test)]
+async fn test_reset_dataset_with_2revisions_drop_last() {
+    let harness = ResetTestHarness::new();
+    let test_case = harness.a_chain_with_2_blocks().await;
+
+    let current_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(test_case.hash_polling_source_block, current_head);        
+
+    let result = harness.reset_svc
+        .reset_dataset(
+            &test_case.dataset_handle, 
+            &test_case.hash_seed_block
+        )
+        .await;
+    assert!(result.is_ok());   
+
+    let new_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(test_case.hash_seed_block, new_head);    
+
+    let summary = harness.get_dataset_summary(&test_case.dataset_handle).await;
+    assert_eq!(new_head, summary.last_block_hash);    
+}
+
+#[test_log::test(tokio::test)]
+async fn test_reset_dataset_with_2revisions_without_changes() {
+    let harness = ResetTestHarness::new();
+    let test_case = harness.a_chain_with_2_blocks().await;
+
+    let current_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(test_case.hash_polling_source_block, current_head);        
+
+    let result = harness.reset_svc
+        .reset_dataset(
+            &test_case.dataset_handle, 
+            &test_case.hash_polling_source_block
+        )
+        .await;
+    assert!(result.is_ok());    
+
+    let new_head = harness.get_dataset_head(&test_case.dataset_handle).await;
+    assert_eq!(current_head, new_head);
+
+    let summary = harness.get_dataset_summary(&test_case.dataset_handle).await;
+    assert_eq!(current_head, summary.last_block_hash);
+}
+
+#[test_log::test(tokio::test)]
+async fn test_reset_dataset_to_non_existing_block_fails() {
+    let harness = ResetTestHarness::new();
+    let test_case = harness.a_chain_with_2_blocks().await;
+
+    let a_hash_not_present_in_chain = Multihash::from_multibase_str(
+        "zW1a3CNT52HXiJNniLkWMeev3CPRy9QiNRMWGyTrVNg4hY8"
+    ).unwrap();
+    
+    let result = harness.reset_svc
+    .reset_dataset(
+        &test_case.dataset_handle, 
+        &a_hash_not_present_in_chain
+    )
+    .await;
+    assert!(result.is_err());    
```

Comment:
We should at least check here the rype of error.
I recommend the following error checking style using `assert_matches!` as it provides nice debug info: https://github.com/kamu-data/kamu-cli/blob/master/kamu-core/tests/tests/repos/test_metadata_chain_impl.rs#L35-L38

Reply:
assert_matches!(result, Err(ResetError::BlockNotFound(_)));